stdlib/public/core/StringUTF16View.swift

```diff
@@ -532,9 +532,12 @@ extension String.UTF16View {
     let idx = _utf16AlignNativeIndex(idx)
 
     guard _guts._useBreadcrumbs(forEncodedOffset: idx._encodedOffset) else {
-      // TODO: Generic _distance is still very slow. We should be able to
-      // skip over ASCII substrings quickly
-      return _distance(from: startIndex, to: idx)
+      return _guts.withFastUTF8 { utf8 in
+        let sliced = UnsafeBufferPointer(
+          rebasing: utf8[startIndex._encodedOffset ..< idx._encodedOffset]
```

Comment:
This assumes that `idx` has already been validated to be within bounds, which isn't currently the case. 

(I'm planning to systematically address this in https://github.com/apple/swift/pull/41417, but I haven't got to the UTF-16 view yet!)

Reply:
eep! Thank you